misc/scripts/deploy-edgeX-snaps.sh

```diff
@@ -0,0 +1,169 @@
+#!/bin/bash
+
+timestamp() {
+  local ts=""
+  ts=$(date +"%Y-%m-%dT%T.%3N") # current time
+  echo "$ts -"
+}
+
+channel_type() {
+  local CHANNEL_VAL=""
+
+  if [[ -z "$1" ]]; then
+    echo "Value not set"
+  else
+    case $1 in
+      stable)
+        CHANNEL_VAL="--stable"
+        ;;
+
+      edge)
+        CHANNEL_VAL="--edge"
+        ;;
+
+      beta)
+        CHANNEL_VAL="--beta"
+      ;;
+
+      candidate)
+        CHANNEL_VAL="--candidate"
+      ;;
+    esac
+  fi
+
+  echo "$CHANNEL_VAL"
+}
+
+EF_CHANNEL_VAL=""
+EF_CHANNEL_VAL_SET=false
```

Comment:
I also think this logic is a bit overly complex, you could simply do:

```
EF_CHANNEL="${EF_CHANNEL:-stable}"

if [ -n "$EF_SNAP" ]; then
    sudo snap install ${EF_SNAP} --dangerous
elif [ -n "$EF_CHANNEL" ]; then
    sudo snap install edgexfoundry --channel=${EF_CHANNEL}
fi
```

And the same for app-service-configurable...

Reply:
Simplified the code for both edgexfoundry and app-service-configurable.